### PR TITLE
Unify Abstract operation factory logic

### PR DESCRIPTION
**Pull request by MacondoExpress**
Tests improved in branch B:  [
  'Cypher Union Read Unions simple',
  'Cypher Union Read Unions with filter and limit',
  'Cypher Union Read Unions with missing types',
  'Interface Relationships Filter overriding using onType',
  'Node directive with unions Read Unions',
  'https://github.com/neo4j/graphql/issues/1779 EXISTS should not be within return clause'
]
Regressions in branch B:  []

### CODE REVIEW

**Comment by neo4j-team-graphql** [comment]
Made some changes to the Neo4j GraphQL Toolbox? Or made some changes to the core library perhaps?

Check out your changes at https://neo4j-graphql-toolbox-3954.surge.sh

Use for instance a local dbms hosted on [Neo4j Desktop](https://neo4j.com/download/) or a cloud-hosted [Aura Free](https://neo4j.com/cloud/aura-free/) dbms.

File: packages/graphql/tests/tck/union.test.ts [hunk 1]
```diff
@@ -30,7 +30,7 @@ describe("Cypher Union", () => {
 
     beforeAll(() => {
         typeDefs = gql`
-            union Search = Movie | Genre
+            union Search = Genre | Movie
```
Comment (angrykoala) [inline packages/graphql/tests/tck/union.test.ts:33]: Why this change? Does it change the order of the resulting union in cypher?

**Review by angrykoala — APPROVED** [review]

File: packages/graphql/tests/tck/union.test.ts (hunk 1 above)
Comment (MacondoExpress) [reply packages/graphql/tests/tck/union.test.ts:33]: Yes, the order in which the union subqueries are depends on the parsing order in the Schema Model, I tried to find a consistent order logic in the previous implementation but I failed. I make these small changes in the type definitions to make it easier to detect the actual changes!
